tests/addons/core/test_action_botfront_form.py

```diff
@@ -118,6 +118,32 @@ def required_slots_graph(conjunction="OR", negated=False):
         ],
     }
 
+def required_slots_graph_with_set_slots(conjunction="OR", negated=False):
```

Comment:
don't copy the test -- just change `required_slots_graph` and `test_required_slots`

Reply:
Actually I don't want that because I specifically want to test if everything goes well when a graph does not end with a slot set.

Reply:
Then you can add slotset as a parameter of the existing test. I just don't like having two copies of the same 50+ lines.